specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json

```diff
@@ -1440,6 +1452,67 @@
         }
       }
     },
+    "UpstreamAuthSettings": {
```

Comment:
just want to understand the context, seems signalr has extra layer of model `UpstreamAuthSettings` ontop of `ManagedIdentity` settings, which is different with other services,  seems it's for extensibility? but currently only managedIdentity and none supported.

Reply:
SignalR Service is a fully managed service, one of the functionality is the service can invokes other service using managed identity to acquire AAD token if customer choose to use `ManagedIdentity` in `UpstreamAuthSettings`. At the meanwhile, customer can specify the `target resource`. 
In simple, `ManagedIdentity` is to assign a identity just like other services. And `UpstreamAuthSettings` is to tell the service when to use managed identity to acquire aad token. `UpstreamAuthSettings` in each `upstream` can vary.